### PR TITLE
[chore] add govulncheck

### LINKED ISSUE #390: [security] audit repository tooling
The security SIG is looking to ensure that security tooling is setup consistently across the organization. As a result, we're asking maintainers to ensure the following tools are enabled in each repository:

- [x] CodeQL enabled via [GitHub Actions](https://github.com/open-telemetry/opentelemetry-collector/blob/main/.github/workflows/codeql-analysis.yml)
- [ ] Static code analysis tool (the collector uses govulncheck [https://pkg.go.dev/golang.org/x/vuln] on every build)
- [x] Repository security settings
    - [x] Security Policy ✅
    - [x] Security advisories ✅
    - [x] Private vulnerability reporting ✅
    - [x] Dependabot alerts ✅
    - [x] Code scanning alerts ✅

Parent issue: https://github.com/open-telemetry/sig-security/issues/12

### PR DESCRIPTION
**Pull request by sakshi-1505**
fixes #390

### CODE REVIEW

**Comment by sakshi-1505** [comment]
@pellared please take a look I have also made some revamps around how current lint job was working.

**Comment by sakshi-1505** [comment]
Also, this time I have made sure to run `make lint` & `make precommit` locally

File: Makefile [hunk 1]
```diff
@@ -57,6 +57,9 @@ $(TOOLS)/crosslink: PACKAGE=go.opentelemetry.io/build-tools/crosslink
 CHLOGGEN = $(TOOLS)/chloggen
 $(TOOLS)/chloggen: PACKAGE=go.opentelemetry.io/build-tools/chloggen
 
+GOVULNCHECK = $(TOOLS)/govulncheck
+ $(TOOLS)/govulncheck: PACKAGE=golang.org/x/vuln/cmd/govulncheck
+
 .PHONY: tools
 tools: $(DBOTCONF) $(GOLANGCI_LINT) $(MISSPELL) $(MULTIMOD) $(CROSSLINK) $(CHLOGGEN)
```
Comment (pellared) [inline Makefile:64]: Add `$(GOVULNCHECK)` here

**Comment by pellared** [comment]
Failing https://github.com/open-telemetry/opentelemetry-go-build-tools/actions/runs/6473513242/job/17577051182?pr=417

```
Error: /home/runner/go/pkg/mod/github.com/golangci/golangci-lint@v1.54.2/pkg/golinters/unparam.go:9:2: missing go.sum entry for module providing package mvdan.cc/unparam/check (imported by github.com/golangci/golangci-lint/pkg/golinters); to add:
	go get github.com/golangci/golangci-lint/pkg/golinters@v1.54.2
make: *** [Makefile:39: /home/runner/work/opentelemetry-go-build-tools/opentelemetry-go-build-tools/.tools/golangci-lint] Error 1
```

**Comment by sakshi-1505** [comment]
> Failing https://github.com/open-telemetry/opentelemetry-go-build-tools/actions/runs/6473513242/job/17577051182?pr=417
> 
> ```
> Error: /home/runner/go/pkg/mod/github.com/golangci/golangci-lint@v1.54.2/pkg/golinters/unparam.go:9:2: missing go.sum entry for module providing package mvdan.cc/unparam/check (imported by github.com/golangci/golangci-lint/pkg/golinters); to add:
> 	go get github.com/golangci/golangci-lint/pkg/golinters@v1.54.2
> make: *** [Makefile:39: /home/runner/work/opentelemetry-go-build-tools/opentelemetry-go-build-tools/.tools/golangci-lint] Error 1
> ```

checking now

**Comment by sakshi-1505** [comment]
please check now @pellared

**Review by pellared — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Comment by codeboten** [comment]
Thanks @sakshi-1505!

**Pull request merged**